control/control.go

```diff
@@ -1049,13 +1024,21 @@ func (p *pluginControl) MetricExists(mns core.Namespace, ver int) bool {
 // CollectMetrics is a blocking call to collector plugins returning a collection
 // of metrics and errors.  If an error is encountered no metrics will be
 // returned.
-func (p *pluginControl) CollectMetrics(metricTypes []core.Metric, deadline time.Time, taskID string, allTags map[string]map[string]string) (metrics []core.Metric, errs []error) {
+func (p *pluginControl) CollectMetrics(id string, allTags map[string]map[string]string) (metrics []core.Metric, errs []error) {
 	// If control is not started we don't want tasks to be able to
 	// go through a workflow.
 	if !p.Started {
 		return nil, []error{ErrControllerNotStarted}
 	}
 
+	_, results, _, _, err := p.subscriptionGroups.Get(id)
```

Comment:
The return for subscriptionGroups.Get seems overloaded. If we only ever want a subset of what is returned maybe it should be split out into multiple functions that return what is needed?